Offer.sol

```diff
@@ -177,16 +195,32 @@ contract Offer {
     // Executing this function before the Offer is accepted by the Client
     // makes no sense as this contract has no ether.
     function withdraw() noEther {
-        if (msg.sender != contractor)
+        if (msg.sender != contractor || now < dateOfSignature + payoutFreezePeriod)
             throw;
         uint timeSincelastWithdrawal = now - lastWithdrawal;
         // Calculate the amount using 1 second precision.
         uint amount = (timeSincelastWithdrawal * dailyWithdrawalLimit) / (1 days);
         if (amount > this.balance) {
             amount = this.balance;
         }
-        if (contractor.send(amount))
-            lastWithdrawal = now;
+        var lastWithdrawalReset = lastWithdrawal;
+        lastWithdrawal = now;
```

Comment:
The 2 lines above can be removed. What if `contractor.send(amount)` fails. then you would have set the wrong value.